### PR TITLE
successfully run geometric controller from robot_framework

### PR DESCRIPTION
**Pull request by aliagilbert**

### CODE REVIEW

**Comment by aliagilbert** [comment]
@hardikparwana can you add the interface for `useExternalController(bool mode)` to `robot-framwork-py` or show me (dev) how to do it?

File: src/DASCRobots.cpp [hunk 1]
```diff
@@ -121,6 +126,7 @@ bool DASCRobot::disarm() {
     VehicleCommand msg;
     msg.timestamp = get_current_timestamp();
     msg.param1 = 0.0;
+    msg.param2 = 21196; // set param2 to 21196 to force arm/disarm operation
```
Comment (aliagilbert) [inline src/DASCRobots.cpp:129]: this was needed to force the disarming. not sure why forcing is necessary.

Comment (hardikparwana) [reply src/DASCRobots.cpp:129]: what was the error when this param was not added? did you try arming from QGC? Did that work?

**Comment by hardikparwana** [comment]
> @hardikparwana can you add the interface for `useExternalController(bool mode)` to `robot-framwork-py` or show me (dev) how to do it?

I'll modify the python interface for external controller myself tomorrow.

**Review by hardikparwana — APPROVED** [review]
Indentations are a mess. code is difficult to read because of that. Everything else looks good!  Just correct the indentations in all the files and then merge with master branch.

File: src/example_control_geometric.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	}
+	else if (elapsed <= 15000*1000) {
+	        rover->cmdWorldPosition(0.0, 0.0, 0.0, 0, 0);
+		std::cout << "Elapsed " << elapsed << "; CMD: 0.0" << std::endl;
+	}
+	else{
+		break;
+	}
+
+        rover->useExternalController(true);
```
Comment (hardikparwana) [inline src/example_control_geometric.cpp:60]: Why keep sending use_external_control command? we just need to send it once. or send it in the previous for loop if just doing it as a caution for data packet loss.

File: src/example_control_geometric.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    auto start = rover->get_current_timestamp();
+
+    while(rclcpp::ok()) {
+
+	//rover->cmdWorldPosition(0,0,0.5,0,0);
+	auto now = rover->get_current_timestamp();
+
+        auto elapsed = now - start;
+
+	if (elapsed <= 10000*1000) {
```
Comment (hardikparwana) [inline src/example_control_geometric.cpp:48]: the time is in milliseconds.. so u are checking for 10,000 x 1000 milliseconds = 10,000 seconds????

File: src/DASCRobots.cpp (hunk 1 above)
Comment (dev10110) [reply src/DASCRobots.cpp:129]: Arming was fine. 
Disarming wouldn't occur without this forcing.

File: src/example_control_geometric.cpp (hunk 2 above)
Comment (dev10110) [reply src/example_control_geometric.cpp:60]: Ah good catch. I'll try sending it only once.

File: src/example_control_geometric.cpp (hunk 3 above)
Comment (dev10110) [reply src/example_control_geometric.cpp:48]: Nope. We did the experiment and it seems like timestamps are in us.

**Pull request merged**